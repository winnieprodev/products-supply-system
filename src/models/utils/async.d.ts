export interface Error {
  message: string;
}

export interface Async<T> {
  isLoading: boolean;
  error: Error | undefined;
  data: T | undefined;
}
