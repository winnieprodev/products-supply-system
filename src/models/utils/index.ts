export * from "./async.d";
export * from "./enums";
