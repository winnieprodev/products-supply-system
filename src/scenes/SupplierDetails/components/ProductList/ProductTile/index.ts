export { default as ProductTile } from "./ProductTile";
