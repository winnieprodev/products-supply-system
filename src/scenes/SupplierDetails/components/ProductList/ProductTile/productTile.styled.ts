import { GenericProductIcon } from "components";
import styled, { css } from "styled-components";
import {
  primaryColor,
  secondaryColor,
  secondaryColorLighter,
} from "styles/colors";

interface BlockProps {
  $selected: boolean;
}

export const Block = styled.div<BlockProps>`
  cursor: pointer;
  background: white;
  position: relative;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
  padding: 10px;

  ${({ $selected }) =>
    $selected &&
    css`
      padding: 28px 10px;
      margin-top: -18px;
      background: #f4f4f4;
    `}
`;

export const Content = styled.div`
  position: relative;
  color: #808080;
  background: ${secondaryColorLighter};
  border-radius: 10px;
  position: relative;
  align-items: center;
  justify-content: center;

  &:hover {
    background: ${secondaryColor};
  }
`;

export const Image = styled.img`
  width: 100%;
  max-height: 100%;
  mix-blend-mode: darken;
`;

export const NoImage = styled(GenericProductIcon)`
  font-size: 100px;
  text-align: center;
  display: inherit;
  margin: 0 auto;
  padding: 4vh 0;
  opacity: 0.2;
`;

export const InfoSection = styled.div`
  width: 100%;
  position: absolute;
  display: flex;
  bottom: 0;
  background: rgba(244, 244, 244, 0.5);
  border-radius: 6px;
`;

export const QuantitySection = styled.div`
  flex-grow: 1;
  padding: 10px 0 10px 10px;
`;

export const QuantityDescription = styled.div`
  color: #000000;
  font-size: 10px;
`;

export const SerialSection = styled.div`
  width: 35%;
  padding: 10px 10px 10px 0;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  color: ${primaryColor};
  font-size: 9px;
  text-align: right;
`;

export const Title = styled.div`
  padding: 10px;
  color: #000000;
  font-size: 16px;
  font-weight: bold;
`;
