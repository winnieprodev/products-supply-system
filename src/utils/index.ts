export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
