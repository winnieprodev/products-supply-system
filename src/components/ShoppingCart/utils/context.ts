import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
