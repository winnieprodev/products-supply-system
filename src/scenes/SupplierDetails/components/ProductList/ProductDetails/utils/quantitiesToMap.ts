import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
