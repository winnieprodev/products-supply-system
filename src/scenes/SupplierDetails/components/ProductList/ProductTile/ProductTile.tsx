import { Product } from "models";

import * as Styled from "./productTile.styled";

interface OwnProps {
  onClick: () => void;
  item: Product;
  isSelected: boolean;
}

const ProductTile: React.FC<React.PropsWithChildren<OwnProps>> = ({
  onClick,
  item,
  isSelected,
}) => {
  return (
    <Styled.Block $selected={isSelected} onClick={onClick}>
      <div>
        <Styled.Content>
          {item.image ? <Styled.Image src={item.image} /> : <Styled.NoImage />}
          <Styled.InfoSection>
            <Styled.QuantitySection>
              {item.quantities.map(({ type, description }, index) => (
                <Styled.QuantityDescription key={index}>
                  {type}: {description}
                </Styled.QuantityDescription>
              ))}
            </Styled.QuantitySection>
            {item.code && (
              <Styled.SerialSection>{item.code}</Styled.SerialSection>
            )}
          </Styled.InfoSection>
        </Styled.Content>
        <Styled.Title>{item.title}</Styled.Title>
      </div>
    </Styled.Block>
  );
};

export default ProductTile;
