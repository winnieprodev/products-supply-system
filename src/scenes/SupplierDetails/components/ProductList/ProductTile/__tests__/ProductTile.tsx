import { render } from "@testing-library/react";
import { mockProduct } from "utils";

import ProductTile from "../ProductTile";

describe("ProductTile", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <ProductTile
          item={mockProduct("id1")}
          onClick={jest.fn()}
          isSelected={false}
        ></ProductTile>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with selected = index", () => {
      const { container } = render(
        <ProductTile
          item={mockProduct("id1")}
          onClick={jest.fn()}
          isSelected={true}
        ></ProductTile>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
