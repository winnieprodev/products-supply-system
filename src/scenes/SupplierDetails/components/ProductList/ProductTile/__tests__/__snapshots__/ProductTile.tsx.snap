// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTile Snapshot tests it matches the snapshot 1`] = `
.c0 {
  cursor: pointer;
  background: white;
  position: relative;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
  padding: 10px;
}

.c1 {
  position: relative;
  color: #808080;
  background: #F3F3F3;
  border-radius: 10px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1:hover {
  background: #EEEEEE;
}

.c2 {
  width: 100%;
  max-height: 100%;
  mix-blend-mode: darken;
}

.c3 {
  width: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  bottom: 0;
  background: rgba(244,244,244,0.5);
  border-radius: 6px;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 10px 0 10px 10px;
}

.c5 {
  color: #000000;
  font-size: 10px;
}

.c6 {
  padding: 10px;
  color: #000000;
  font-size: 16px;
  font-weight: bold;
}

<div
  class="c0"
>
  <div>
    <div
      class="c1"
    >
      <img
        class="c2"
        src="src"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            loose
            : 
            12 x 1L
          </div>
        </div>
      </div>
    </div>
    <div
      class="c6"
    >
      Whipping Cream 35% 1L
    </div>
  </div>
</div>
`;

exports[`ProductTile Snapshot tests it matches the snapshot with selected = index 1`] = `
.c0 {
  cursor: pointer;
  background: white;
  position: relative;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
  padding: 10px;
  padding: 28px 10px;
  margin-top: -18px;
  background: #f4f4f4;
}

.c1 {
  position: relative;
  color: #808080;
  background: #F3F3F3;
  border-radius: 10px;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1:hover {
  background: #EEEEEE;
}

.c2 {
  width: 100%;
  max-height: 100%;
  mix-blend-mode: darken;
}

.c3 {
  width: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  bottom: 0;
  background: rgba(244,244,244,0.5);
  border-radius: 6px;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 10px 0 10px 10px;
}

.c5 {
  color: #000000;
  font-size: 10px;
}

.c6 {
  padding: 10px;
  color: #000000;
  font-size: 16px;
  font-weight: bold;
}

<div
  class="c0"
>
  <div>
    <div
      class="c1"
    >
      <img
        class="c2"
        src="src"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            loose
            : 
            12 x 1L
          </div>
        </div>
      </div>
    </div>
    <div
      class="c6"
    >
      Whipping Cream 35% 1L
    </div>
  </div>
</div>
`;
